crates/forge_provider/src/retry.rs

```diff
@@ -97,6 +98,15 @@ fn is_event_transport_error(error: &anyhow::Error) -> bool {
         .is_some_and(|e| matches!(e, reqwest_eventsource::Error::Transport(_)))
 }
 
+fn is_json_parse_error(error: &anyhow::Error) -> bool {
+    // ToolCallMissingName is also a json parse error,
+    // This happens in case when the response is empty JSON.
+    error
+        .downcast_ref::<Error>()
+        .is_some_and(|e| matches!(e, Error::InvalidJson(_) | Error::ToolCallMissingName))
```

Comment:
Check for response and if response is an empty error then we can consider retrying.